### ISSUE TITLE
The change for #1148 breaks migration

### ISSUE THREAD

**Issue by jopit** [eclipse/codewind#1558]
The change for https://github.com/eclipse/codewind/issues/1148  breaks migration if an older version of codewind is already running.

I installed and started codewind 0.6.1, then started the latest version of vscode. When I tried to migrate, I got an error, possibly because there is no docker-compose file in the ~/.codewind folder.

Note that cwctl and therefore the IDEs think that codewind is not running when in fact it is, because the new cwctl is looking for containers whose names start with eclipse/.
```
cwctl is available at /Users/jsp/.codewind/0.7.0/cwctl
==> Run cwctl --json --insecure connections list
{"schemaversion":1,"connections":[{"id":"local","label":"Codewind local connection","url":"","auth":"","realm":"","clientid":"","username":""}]}
==> End cwctl --json --insecure connections list
==> Run cwctl --json --insecure status
{"status":"stopped","installed-versions":["0.6.1"]}
==> End cwctl --json --insecure status
==> Run cwctl --json --insecure status
{"status":"stopped","installed-versions":["0.6.1"]}
==> End cwctl --json --insecure status
==> Run cwctl --json --insecure status
{"status":"stopped","installed-versions":["0.6.1"]}
==> End cwctl --json --insecure status
==> Run cwctl --json --insecure status
{"status":"stopped","installed-versions":["0.6.1"]}
==> End cwctl --json --insecure status
==> Run cwctl --json remove -t 0.6.1
Removing Codewind docker images..
System architecture is:  amd64
Host operating system is:  darwin
Please wait whilst images are removed...  
2019/12/16 16:46:25 DOCKER_COMPOSE_ERROR[101]: exit status 1. docker-compose remove command failed to execute correctly
==> End cwctl --json remove -t 0.6.1
```

**Comment by ssh24** [comment]
I ran into this as well on Windows. Basically, you need to run `docker rm -f codewind-pfe` to manually stop PFE, and then start cwctl which will create the `docker-compose` file in `/.codewind` directory. I think this needs to be documented somewhere unless there is a more elegant way of doing this.

**Comment by stalleyj** [comment]
I'm sure we can make cwctl a bit cleverer to handle this situation

**Comment by stalleyj** [comment]
@jopit I can't seem to recreate this problem any longer. I installed the 0.6.1 plugin and once everything was up and running I installed a 0.8.0 version and everything seemed to work as expected. Is there something I'm doing differently ?  I thought I understood the issues of the difference in image names and changes around docker compose but can't get it to fail.

**Comment by jopit** [comment]
Did you delete the ~/.codewind folder before installing 0.6.1 to ensure there is no docker-compose.yaml file?

**Comment by stalleyj** [comment]
Yes, I did delete the .codewind directory and started with a completely clean environment.

**Comment by stalleyj** [comment]
I can make this fail by using cwctl commands but I am still not able to get a failure when using VSCode.  I had 0.6.1 running then tried starting 0.8.0 and the plugin shows the output below.  I don't see the **remove** command being called which is what was originally failing. I would like to understand the scenario of when the plugin calls remove @tetchel ?

[Extension Host] [DBUG: 13:32:05.781 CLIWrapper.js:192]: Successfully ran CLI command cwctl --json --insecure status, Output was:
{"status":"stopped","installed-versions":["0.8.0","latest","0.7.0","0.6.1"]}
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [INFO: 13:32:10.507 StartCodewindCmd.js:36]: Connect Local Codewind Cmd
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [INFO: 13:32:10.508 CLIWrapper.js:129]: Running CLI command: cwctl --json --insecure status
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [DBUG: 13:32:12.013 CLIWrapper.js:192]: Successfully ran CLI command cwctl --json --insecure status, Output was:
{"status":"stopped","installed-versions":["0.8.0","latest","0.7.0","0.6.1"]}
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [INFO: 13:32:12.014 CLIWrapper.js:129]: Running CLI command: cwctl --json --insecure status
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [DBUG: 13:32:13.416 CLIWrapper.js:192]: Successfully ran CLI command cwctl --json --insecure status, Output was:
{"status":"stopped","installed-versions":["0.8.0","latest","0.7.0","0.6.1"]}
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [INFO: 13:32:13.417 CLILifecycleWrapper.js:135]: Ready to install and start Codewind 0.8.0
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [INFO: 13:32:13.417 CLILifecycleWrapper.js:138]: Codewind is not running
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [INFO: 13:32:13.417 CLILifecycleWrapper.js:177]: Codewind 0.8.0 is already installed
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [DBUG: 13:32:13.417 LocalCodewindManager.js.setState():138]: Local Codewind state changing from Stopped to Starting
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [INFO: 13:32:13.418 CLIWrapper.js:129]: Running CLI command: cwctl --json start -t 0.8.0
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:250 [Extension Host] [DBUG: 13:32:31.535 CLIWrapper.js:192]: Successfully ran CLI command cwctl --json start -t 0.8.0, Output was:
Debug: false
==> created file /Users/stalleyj/.codewind/0.8.0/Users/stalleyj/.codewind/docker-compose.yaml
==> environment structure written to /Users/stalleyj/.codewind/docker-compose.yaml
System architecture is:  amd64
Host operating system is:  darwin
Attempting to find available port
Port 10001 Available
Please wait while containers initialize...  
Recreating codewind-performance ... 
[1A[2K
Recreating codewind-performance ... [32mdone[0m
[1BRecreating codewind-pfe         ... 
[1A[2K
Recreating codewind-pfe         ... [32mdone[0m
[1BWaiting for Codewind to start
..
HTTP Response Status: 200 OK
Codewind successfully started on http://127.0.0.1:10001

**Comment by tetchel** [comment]
After the prompt which says "you have x version installed, but y version is required" and the user confirms to upgrade, all the old versions are removed before proceeding to install the new version.

**Comment by stalleyj** [comment]
Finally, I can recreate this and see that the remove command fails but I'm only aware from looking at the logs. 0.8.0 is running and connected and I don't think a user would be aware that anything went wrong.

**Comment by stalleyj** [comment]
@jopit As far as I can see, the only consequence to a user is that some Codewind images will be left on their machine which I don't think justifies the effort required to fix this. Are you happy to close ?

**Comment by jopit** [comment]
Hi @stalleyj I think if the  `cwctl remove` command isn't removing the images like it's supposed to, it should be fixed.  I don't think this issue qualifies as hot anymore, so I will remove the hot tag.

**Comment by jopit** [comment]
Last week I tried the same scenario starting from Codewind 0.7.0, and it also did not remove the images, so it's not just a problem with 0.6.x

**Comment by malincoln** [comment]
@tobespc I'm removing the 0.8.0 release label since that has past.  Pls update release once you know when it will be targeted.  Thanks

**Comment by tobespc** [comment]
Moving this to icebox.  With the assumption that everyone will eventually be past the version with issues then this will not be a problem going forwards
